.github/workflows/py-workflow.yml

```
@@ -0,0 +1,568 @@
+      prod_publish_repos:
+        required: false
+        type: string
+        default: '["pypi-edge"]'
+      PYTHON_VERSION:
+        required: true
+        type: string
+        default: "3.8.10"
+    secrets:
+      auto_commit_user:
+        required: false # only when deploy true
+      auto_commit_mail:
+        required: false # only when deploy true
+      auto_commit_password:
+        required: false
+      registry_user:
+        required: true
+      registry_password:
+        required: true
+      nexus_publisher_user:
+        required: true
+      nexus_publisher_password:
+        required: true
+      gh_token:
+        required: true
+      sonar_token:
+        required: true
+
+jobs:
+  RefreshSource:
+    if: ${{ ! inputs.deploy }}
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: docker://chinthakagodawita/autoupdate-action:v1
+        continue-on-error: true
+        env:
+          GITHUB_TOKEN: '${{ secrets.auto_commit_password }}'
+          MERGE_MSG: "Branch was auto-updated."
+
+  Pre-build:
+    if: ${{ !inputs.release }}
+    runs-on: ubuntu-22.04
+    container:
+      image: registry.cloud.mov.ai/qa/py-buildserver:v3.0.4
+      credentials:
+        username: ${{secrets.registry_user}}
+        password: ${{secrets.registry_password}}
+    outputs:
+      branch: ${{ steps.var_branch.outputs.branch }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        submodules: recursive
+
+    - name: Set branch output
+      if: ${{ inputs.deploy }}
+      id: var_branch
+      run: |
+        echo "Branch name is: ${GITHUB_REF#refs/heads/}"
+        echo "branch=${GITHUB_REF#refs/heads/}" >> $GITHUB_OUTPUT
+
+  Pre-commit-38:
+    if: ${{ inputs.pre_commit && startsWith(inputs.PYTHON_VERSION, '3.8') }}
+    runs-on: ubuntu-22.04
+    needs: [ Pre-build ]
+    container:
+      image: registry.cloud.mov.ai/qa/py-buildserver:v3.0.4
+      credentials:
+        username: ${{secrets.registry_user}}
+        password: ${{secrets.registry_password}}
+    steps:
+      - name: Install dependencies and package
+        run: |
+          python3 -m pip install pip pre-commit--upgrade
+          python3 -m pip install -i https://artifacts.cloud.mov.ai/repository/pypi-integration/simple --extra-index-url https://pypi.org/simple -r build-requirements.txt
+          python3 -m pip install -i https://artifacts.cloud.mov.ai/repository/pypi-integration/simple --extra-index-url https://pypi.org/simple .
+
+      - name: Run pre-commit
+        run: |
+          git config --global --add safe.directory $(pwd)
+          python -m pre_commit run --show-diff-on-failure --color=always --all-files
+
+  Pre-commit-other:
+    if: ${{ inputs.pre_commit && !startsWith(inputs.PYTHON_VERSION, '3.8') }}
```

This seems to be missing in v2 too:
```suggestion
    if: ${{ !inputs.release && inputs.pre_commit && !startsWith(inputs.PYTHON_VERSION, '3.8') }}
```